backend/src/main/kotlin/database/PostgreRepository.kt

```diff
@@ -0,0 +1,164 @@
+package database
+
+import entities.Season
+import entities.SeasonHalf.Companion.toSeasonHalf
+import entities.Store
+import org.jetbrains.exposed.sql.*
+import org.jetbrains.exposed.sql.`java-time`.date
+import org.jetbrains.exposed.sql.transactions.TransactionManager
+import org.jetbrains.exposed.sql.transactions.transaction
+import java.sql.SQLException
+import java.time.Year
+
+class PostgreRepository(private val database: Database) : Repository {
+    private object StoreTable : Table("store") {
+        val id = long("id")
+        val code = varchar("code", 3000).nullable()
+        val description = varchar("description", 3000).nullable()
+        val name = varchar("name", 100)
+        val openingDate = date("openingDate").nullable()
+        val storeType = varchar("storeType", 100).nullable()
+        val specialField1 = varchar("specialField1", 1500).nullable()
+        val specialField2 = varchar("specialField2", 1500).nullable()
+
+        override val primaryKey = PrimaryKey(id)
+    }
+
+    private object SeasonTable : Table("season") {
+        val half = varchar("half", 2)
+        val year = integer("year")
+        override val primaryKey = PrimaryKey(half, year)
+    }
+
+    private object StoreSeasonTable : Table("stores_seasons") {
+        val storeId = long("storeId").references(StoreTable.id, onDelete = ReferenceOption.CASCADE)
+        val half = varchar("half", 2)
+        val year = integer("year")
+
+    }
+
+    override fun importStores(stores: List<Store>): Unit = transaction(database) {
+        StoreTable.batchInsert(stores) { store ->
+            this[StoreTable.id] = store.id
+            this[StoreTable.code] = store.code
+            this[StoreTable.description] = store.description
+            this[StoreTable.name] = store.name
+            this[StoreTable.openingDate] = store.openingDate
+            this[StoreTable.storeType] = store.storeType
+        }
+    }
+
+    override fun getStores() = transaction(database) {
+        StoreTable.selectAll().map {
+            Store(
+                id = it[StoreTable.id],
+                code = it[StoreTable.code],
+                description = it[StoreTable.description],
+                name = it[StoreTable.name],
+                openingDate = it[StoreTable.openingDate],
+                storeType = it[StoreTable.storeType],
+                specialField1 = it[StoreTable.specialField1],
+                specialField2 = it[StoreTable.specialField2],
+            )
+        }
+    }
+
+    override fun updateStores(stores: List<Store>): Unit = transaction(database) {
+        for (store in stores) {
+            StoreTable.update({ StoreTable.id eq store.id }) {
```

Comment:
Is there a reason not to batch updates as well?

Reply:
I couldn't find a batch update statement in [Exposed queries](https://github.com/JetBrains/Exposed/blob/e710642a20e48d31274081a27cc79cbe1a53c637/exposed-core/src/main/kotlin/org/jetbrains/exposed/sql/Queries.kt)